libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1870,7 +1870,12 @@ void ExpressionCompiler::appendExternalFunctionCall(
 				break;
```

Comment:
Further up: Should we also use `decodingType` to query for "dynamically encoded"?

Reply:
I have the feeling that a couple of these flags and parameters in the AST are not entirely well-defined and/or not consistently used at the moment...

Reply:
A dynamically sized storage array will have ``isDynamicallyEncoded() == true``, but ``decodingType`` will be ``IntegerType(256)``, which will have ``isDynamicallyEncoded() == false``, so it won't be the same at least.

Reply:
However, ``calldataEncodingSize()`` and ``decodingType()->calldataEncodingSize()`` will also for example be different for a fixed size storage array I think - not sure why that's not a problem for the change in this PR.

Reply:
You are right, it is fine as it is.